bot/exts/halloween/candy_collection.py

```diff
@@ -72,38 +85,36 @@ async def on_reaction_add(self, reaction: discord.Reaction, user: discord.Member
 
         # if its not a candy or skull, and it is one of 10 most recent messages,
         # proceed to add a skull/candy with higher chance
-        if str(reaction.emoji) not in ('\N{SKULL}', '\N{CANDY}'):
-            if message.id in await self.ten_recent_msg():
+        if str(reaction.emoji) not in (EMOJIS['SKULL'], EMOJIS['CANDY']):
+            recent_message_ids = map(
+                lambda m: m.id,
+                await self.hacktober_channel.history(limit=10).flatten()
+            )
+            if message.id in recent_message_ids:
                 await self.reacted_msg_chance(message)
             return
 
-        for react in self.msg_reacted:
-            # check to see if the message id of a message we added a
-            # reaction to is in json file, and if nobody has won/claimed it yet
-            if react['msg_id'] == message.id and react['won'] is False:
-                react['user_reacted'] = user.id
-                react['won'] = True
-                try:
-                    # if they have record/candies in json already it will do this
-                    user_records = self.get_candyinfo[user.id]
-                    if str(reaction.emoji) == '\N{CANDY}':
-                        user_records['record'] += 1
-                    if str(reaction.emoji) == '\N{SKULL}':
-                        if user_records['record'] <= 3:
-                            user_records['record'] = 0
-                            lost = 'all of your'
-                        else:
-                            lost = random.randint(1, 3)
-                            user_records['record'] -= lost
-                        await self.send_spook_msg(message.author, message.channel, lost)
-
-                except KeyError:
-                    # otherwise it will raise KeyError so we need to add them to file
-                    if str(reaction.emoji) == '\N{CANDY}':
-                        print('ok')
-                        d = {"userid": user.id, "record": 1}
-                        self.candy_json['records'].append(d)
-                await self.remove_reactions(reaction)
+        if message.id in self.candy_messages and str(reaction.emoji) == EMOJIS['CANDY']:
+            self.candy_messages.remove(message.id)
+            prev_record = self.candy_records.get(str(message.author.id), 0)
+            self.candy_records[str(message.author.id)] = prev_record + 1
```

Comment:
```suggestion
            prev_record = self.candy_records.get(str(user.id), 0)
            self.candy_records[str(user.id)] = prev_record + 1
```
This previously added the candies to the person who sent the message and not to the person who clicked the candy. `user.id` will get the ID of the person who reacted to the message, while `message.author.id` will get the ID of the person who sent the message. An easy fix once you spot the issue.